### PR TITLE
Add auto-generated Fish shell lncli-completions

### PR DESCRIPTION
**Pull request by torkelrogstad**
`urfave/cli` added functionality for generating Fish shell tab-completions a few releases ago. We bump the dependency version to get access to this functionality, and add a new, hidden subcommand that generates Fish shell compatible tab-completions. 

This PR supersedes #2844

### CODE REVIEW

File: cmd/lncli/commands.go [hunk 1]
```diff
@@ -2656,6 +2658,21 @@ func updateChannelPolicy(ctx *cli.Context) error {
...
 
+var fishCompletionCommand = cli.Command{
+	Name:   "fish-completion",
+	Hidden: true,
+	Action: func(c *cli.Context) error {
+		completion, err := c.App.ToFishCompletion()
+		if err != nil {
+			return err
+		}
+		// we don't want to suggest files, so we add this first line to the completions
```
Comment (bhandras) [inline cmd/lncli/commands.go:2669]: Style nit: pls make the comment a sentence (capitalize, punctuation). Also lines shouldn't exceed 80 chars.

File: cmd/lncli/commands.go [hunk 2]
```diff
@@ -1135,10 +1135,12 @@ var createCommand = cli.Command{
 			Name: "multi_backup",
 			Usage: "a hex encoded multi-channel backup obtained " +
 				"from exportchanbackup",
+			TakesFile: true,
```
Comment (bhandras) [inline cmd/lncli/commands.go:1138]: Could split the PR into two commits, one bumping the version and adding these flags, the other one for the `fish-completion`.

File: cmd/lncli/commands.go [hunk 3]
```diff
@@ -2656,6 +2658,21 @@ func updateChannelPolicy(ctx *cli.Context) error {
...
+var fishCompletionCommand = cli.Command{
+	Name:   "fish-completion",
+	Hidden: true,
+	Action: func(c *cli.Context) error {
+		completion, err := c.App.ToFishCompletion()
+		if err != nil {
+			return err
+		}
+		// we don't want to suggest files, so we add this first line to the completions
+		completion = fmt.Sprintf("complete -c %q -f \n", c.App.Name) + completion
```
Comment (bhandras) [inline cmd/lncli/commands.go:2670]: Could be a one liner: `fmt.Printf("complete -c %q -f \n%v", c.App.Name, completion)`.

**Comment by torkelrogstad** [comment]
Fixed your comments and split into two commits.

**Comment by torkelrogstad** [comment]
Rebased

**Review by bhandras — APPROVED** [review]
LGTM, thanks @torkelrogstad!

Please update `docs/release-notes/release-notes-0.15.0.md` as well.

File: cmd/lncli/commands.go [hunk 4]
```diff
@@ -2127,6 +2127,21 @@ func updateChannelPolicy(ctx *cli.Context) error {
...
 
+var fishCompletionCommand = cli.Command{
+	Name:   "fish-completion",
+	Hidden: true,
+	Action: func(c *cli.Context) error {
+		completion, err := c.App.ToFishCompletion()
+		if err != nil {
+			return err
+		}
+		// We don't want to suggest files, so we add this
```
Comment (guggero) [inline cmd/lncli/commands.go:2138]: nit: add empty line before comment.

File: cmd/lncli/main.go [hunk 5]
```diff
@@ -272,14 +272,16 @@ func main() {
...
+			TakesFile: true,
 		},
 		cli.StringFlag{
-			Name:  "tlscertpath",
-			Value: defaultTLSCertPath,
-			Usage: "The path to lnd's TLS certificate.",
+			Name:      "tlscertpath",
+			Value:     defaultTLSCertPath,
+			Usage:     "The path to lnd's TLS certificate.",
+			TakesFile: true,
```
Comment (guggero) [inline cmd/lncli/main.go:284]: Oh, this `TakesFile` flag is pretty useful!

File: docs/INSTALL.md [hunk 6]
```diff
@@ -241,6 +241,11 @@ To check that `lnd` was installed properly run the following command:
 This command requires `bitcoind` (almost any version should do) to be available
 in the system's `$PATH` variable. Otherwise some of the tests will fail.
 
+**Tab completion for `lncli`**
```
Comment (guggero) [inline docs/INSTALL.md:244]: nit: Call this `Command-line completion` since the tabulator key is just the default key binding for it?

**Review by guggero — COMMENTED** [review]
Missing an entry in the `0.15.0` release notes. Other than that and the two nits, LGTM :tada:

**Comment by torkelrogstad** [comment]
Fixed nits, updated release notes

**Review by guggero — APPROVED** [review]
Needs a rebase, other than that LGTM :tada:

**Comment by torkelrogstad** [comment]
Rebased.

**Review by guggero — APPROVED** [review]

**Review by bhandras — APPROVED** [review]
LGTM :fish:

**Pull request merged**